src/Engine/ProtoCore/Lang/Replication/ReplicationInstruction.cs

```diff
@@ -44,5 +45,22 @@ public override string ToString()
             }
 
         }
+
+        public Boolean Equals(ReplicationInstruction oldOption) {
+
+            if (this.Zipped == oldOption.Zipped)
+            {
+                if (this.ZipIndecies == null && oldOption.ZipIndecies == null)
+                    return true;
+
+                if (this.ZipIndecies != null && oldOption.ZipIndecies != null)
```

Comment:
I'm not getting something here - can you add a comment - but like I said above, I'm confused why we're using sequence equals or something like that?

Reply:
@reddyashish explained this is an optimization.

Reply:
Fastest way to compare elements of both the lists:
https://stackoverflow.com/questions/12795882/quickest-way-to-compare-two-list

I have added comments to the code.